portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,61 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<p>Hello Nick!</p>");
-  }
+    Game game = new Game();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+        String json = new Gson().toJson(game);
+
+        System.out.println(json);
+        response.getWriter().println(json);
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // If the user sends another POST request after the game is over, then start a new game.
+
+        game.players.first = request.getParameter("p1");
+        game.players.second = request.getParameter("p2");
+        // Get the input from the form.
+
+        // Redirect back to the HTML page.
+        response.sendRedirect("/game.html");
+    }
+
+}
+
+class Pair{
```

Comment:
Is there a reason this is called a `Pair` and not a `Move`? If I'm understanding this correctly this is a class for moves where first is where the piece starts and second is where the piece ends, right?

Reply:
I looked a little lower, it looks like it is also used to hold player names. Even though they are functionally the same, it seems a bit weird to me to use the Pair class for two different use cases like this.